### PR TITLE
Add nest components of workflow to snakemake

### PR DESCRIPTION
**Pull request by ethanwhite**
Splits the nest workflow into a per flight detection, per flight processing, and results combination step.
This requires a custom input function for grouping flights into site-year combinations.
It also involves additional code related to fixing issues that arise from processing each in smaller chunks, including properly writing and reading empty shape files.

### CODE REVIEW

**Comment by ethanwhite** [comment]
@henrykironde This is ready for review. I've run the entire workflow on the real data on the HPG and checked in the end products in one of the Shiny apps, so everything works. The key thing that makes this possible in snakemake is the custom input function here:

https://github.com/weecology/EvergladesTools/blob/cf8b03e07605d35bf7ce1f072475b86ab3a32bff/Zooniverse/Snakefile#L46-L61

It creates the lists of input files for each site-year combination by filtering the full list of flights generated by `glob_wildcards` to only the ones that match a particular site-year combination. Happy to walk you through it if useful.

**Comment by henrykironde** [comment]
Let us meet and talk about the development setup you have on your side. I am getting some errors, I believe I am missing something thing.

```python
AttributeError: 'NoneType' object has no attribute 'to_wkt'
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 540/540 [00:53<00:00, 10.07it/s]
/blue/ewhite/henrysenyondo/conda/envs/Zooniverse/lib/python3.10/site-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
Traceback (most recent call last):
  File "/blue/ewhite/everglades/EvergladesTools/Zooniverse/predict.py", line 78, in <module>
    result = run(proj_tile_path=path, checkpoint_path=checkpoint_path, savedir=savedir)
  File "/blue/ewhite/everglades/EvergladesTools/Zooniverse/predict.py", line 58, in run
    projected_boxes = project(proj_tile_path, boxes)
  File "/blue/ewhite/everglades/EvergladesTools/Zooniverse/predict.py", line 35, in project
    boxes.crs = dataset.crs.to_wkt()
AttributeError: 'NoneType' object has no attribute 'to_wkt'
```

**Comment by ethanwhite** [comment]
Happy to find some time to chat today. That error is unrelated to this PR (except that we are actually running and syncing everything now) because it's coming from `predict.py`. It is related to #130, probably due to the file names I'd changed getting overwritten by the sync.

File: Zooniverse/Snakefile [hunk 1]
```diff
@@ -31,8 +37,42 @@ rule predict_birds:
...
+        if site == wildcards.site and year == wildcards.year:
+            flight_path = os.path.join(basepath, year, site, f"{flight}_projected.shp")
+            flights_in_year_site.append(flight_path)
+    return flights_in_year_site
+
+rule detect_nests:
+    input:
+        flights_in_year_site
+    output:
+        "/blue/ewhite/everglades/detected_nests/{year}/{site}/{site}_{year}_detected_nests.shp"
```
Comment (henrykironde) [inline Zooniverse/Snakefile:59]: Can we move the paths to the top of the file? 
eg
DETECTED_NEST = "/blue/ewhite/everglades/detected_nests/{year}/{site}/{site}_{year}_detected_nests.shp"

**Review by henrykironde — COMMENTED** [review]
Looks like the pipeline is working. I have tested this but still having file errors. If we can move the file to the top, I could go ahead and make some dummy files to further test the PR.

File: Zooniverse/Snakefile (hunk 1 above)
Comment (ethanwhite) [reply Zooniverse/Snakefile:59]: No, that will break the wildcard system. I'm free to Zoom for a bit and look at what you're seeing.

**Comment by henrykironde** [comment]
@ethanwhite Thanks for the work on this.

**Comment by ethanwhite** [comment]
My pleasure. Thanks @henrykironde!

**Pull request merged**